src/Sulu/Bundle/AdminBundle/Resources/js/components/CircleSelection/withPercentageValues.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+import React from 'react';
+import type {ComponentType} from 'react';
+import {buildHocDisplayName} from '../../utils/react';
+import type {SelectionData} from './types';
+
+type Props = {
+    containerHeight: number,
+    containerWidth: number,
+    maxRadius: number | typeof undefined,
+    minRadius: number | typeof undefined,
+    onChange: (value: ?SelectionData) => void,
+    usePercentageValues: boolean,
+    value: SelectionData | typeof undefined,
+};
+
+export default function withPercentageValues(Component: ComponentType<*>) {
+    class WithPercentageValuesComponent extends React.Component<Props> {
+        wrappedComponent = Component;
+
+        static defaultProps = {
+            maxRadius: undefined,
+            minRadius: undefined,
+            usePercentageValues: false,
+        };
+
+        handleChange = (value: ?SelectionData) => {
+            const {containerHeight, containerWidth, onChange} = this.props;
+
+            if (!value) {
+                onChange(value);
+
+                return;
+            }
+
+            onChange({
+                ...value,
+                left: value.left / containerWidth,
+                top: value.top / containerHeight,
+                radius: value.radius / containerWidth,
+            });
+        };
+
+        getTransformedMaxRadius = () => {
+            const {containerWidth, maxRadius} = this.props;
+
+            if (!maxRadius) {
+                return maxRadius;
+            }
+
+            return maxRadius * containerWidth;
+        };
+
+        getTransformedMinRadius = () => {
+            const {containerWidth, minRadius} = this.props;
+
+            if (!minRadius) {
+                return minRadius;
+            }
+
+            return minRadius * containerWidth;
+        };
+
+        getTransformedValue = () => {
+            const {containerHeight, containerWidth, value} = this.props;
+
+            if (!value) {
+                return value;
+            }
+
+            return {
+                ...value,
+                left: value.left * containerWidth,
+                top: value.top * containerHeight,
+                radius: value.radius * containerWidth,
+            };
+        };
+
+        render() {
+            const {usePercentageValues} = this.props;
+
+            if (!usePercentageValues) {
```

Comment:
I've imagine it a bit differently :see_no_evil: I thought that the component is only wrapped with this HOC if we want it to use percentage values, but then it would always do that, and does not need that prop anymore... But maybe I was to naive, and this is not possible...

Reply:
I played a few scenarios in my head and wasn't able to find a "nice" solution ... Best would be passing `containerHeight`, `containerWidth` and `usePercentageValue` as additional arguments to `withPercentageValues`, but that would require me creating the `withPercentageValues` instance in `CircleSelection.render()`, because I need to have access to the `CircleSelection`'s props and that's not really good, since it would always completely recreate the wrapped component on render.